src/cloudai/_core/registry.py

```
@@ -190,7 +192,7 @@ def add_agent(self, name: str, value: Type[BaseAgent]) -> None:
         Raises:
             ValueError: If the agent implementation already exists.
         """
-        if name in self.agents_map:
+        if name in self.agents_map or name in self.agent_entrypoints_map:
             raise ValueError(f"Duplicating implementation for '{name}', use 'update()' for replacement.")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)